pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -462,6 +463,8 @@ func (s *DiscoveryServer) edsUpdate(clusterID, serviceName string, namespace str
 			if svcShards == 0 {
 				delete(s.EndpointShardsByService[serviceName], namespace)
 			}
+			adsLog.Infof("Full push, service %s has no endpoints", serviceName)
+			s.ConfigUpdate(&model.PushRequest{Full: true})
```

Comment:
Can you also set TargetNamespace?

Reply:
Huh, I could have sworn this did have this set when I looked earlier... @XanderStrike did you remove it?

Reply:
I added it as a premature optimization, but unfortunately I discovered while manually testing that with namespace specified it doesn't actually send an update. Not sure why. If you like I can dig into that further I'm happy to do it.